pkg/controllers/managementuser/resourcequota/resource_quota_sync.go

```diff
@@ -119,69 +118,84 @@ func (c *SyncController) CreateResourceQuota(ns *corev1.Namespace) (runtime.Obje
 		return ns, err
 	}
 
-	projectLimit, _, err := getProjectResourceQuotaLimit(ns, c.ProjectLister)
-	if err != nil {
-		return ns, err
-	}
-
-	var quotaSpec *corev1.ResourceQuotaSpec
-	if projectLimit != nil {
-		quotaSpec, err = c.getNamespaceResourceQuota(ns)
+	var nsResourceQuota v32.NamespaceResourceQuota
+	if projectLimit, _, err := getProjectResourceQuotaLimit(ns, c.ProjectLister); err == nil && projectLimit != nil {
+		nsResourceQuota, err = c.getNamespaceResourceQuota(ns)
 		if err != nil {
 			return ns, err
 		}
 	}
 
-	quotaToUpdate, err := c.getResourceQuotaToUpdate(ns)
+	quotaToUpdate, err := c.getResourceQuotaToUpdate(ns, &nsResourceQuota)
```

Comment:
Good point! When do we need to look at project's total quota? I think we don't need them (at least in this method). We do care about project's default, but those are fetched inside `getResourceQuotaToUpdate`. So I will remove the code that retrieves total quota - a call to `getProjectResourceQuotaLimit`.

Reply:
I would tend to agree with you - not sure why there was an if check here in the first place. I think it didn't end up affecting anything because the merged code re-fetched it anyways, but I'm not 100% sure on that.

Reply:
Resolving - changes made.